client/app/components/SearchBar/SearchBar.js

```diff
@@ -31,14 +156,24 @@ class SearchBar extends Component {
       type: 'search',
       className: css.keywordInput,
       placeholder: keywordPlaceholder,
-      onChange: (e) => this.setState({ keyword: e.target.value }), // eslint-disable-line react/no-set-state
+      ref: (c) => {
+        this.keywordInput = c;
+      },
     });
     const locationInput = input({
       type: 'search',
       className: css.locationInput,
       placeholder: locationPlaceholder,
       autoComplete: 'off',
-      onChange: (e) => this.setState({ location: e.target.value }), // eslint-disable-line react/no-set-state
```

Comment:
The Google Map Autocomplete sets the value of the input, which doesn't trigger `onChange` and results in the state not being in sync with what the user has selected. Therefore we check the input value in the `handleSubmit` method when the value is actually needed.